src/components/smart_objects/src/object_schema_item.cc

```diff
@@ -157,6 +158,31 @@ errors::eType CObjectSchemaItem::validate(
   return errors::OK;
 }
 
+bool CObjectSchemaItem::filterInvalidEnums(
+    SmartObject& Object,
+    const utils::SemanticVersion& MessageVersion,
+    rpc::ValidationReport* report__) {
+  for (const auto& key : Object.enumerate()) {
```

Comment:
Why don't you check  `Object.getType` ?

Reply:
Ah, I added the `GetType` function for use with my original (and simpler) implementation of this, it didn't turn out to be necessary in the final implementation, so I can remove that function.

Reply:
I ended up using it in this class to filter out single invalid enums from the object